src/state/slices/marketDataSlice/marketDataSlice.tsx

```diff
@@ -163,25 +168,39 @@ export const marketApi = createApi({
       },
       keepUnusedDataFor: 5, // Invalidate cached asset market data after 5 seconds.
     }),
-    findPriceHistoryByAssetId: build.query<HistoryData[] | null, FindPriceHistoryByAssetIdArgs>({
-      queryFn: async (args, { dispatch }) => {
-        const { assetId, timeframe } = args
-        if (shouldIgnoreAsset(assetId)) return { data: [] }
-        try {
-          const data = await getMarketServiceManager().findPriceHistoryByAssetId({
-            timeframe,
-            assetId,
-          })
-          const payload = { args, data }
-          dispatch(marketData.actions.setCryptoPriceHistory(payload))
-          return { data }
-        } catch (e) {
-          const error = {
-            data: `findPriceHistoryByAssetId: error fetching price history for ${assetId}`,
-            status: 400,
-          }
-          return { error }
-        }
+    findPriceHistoryByAssetIds: build.query<null, FindPriceHistoryByAssetIdArgs>({
+      queryFn: async function findPriceHistoryByAssetIds(args, { dispatch }) {
+        const { assetIds, timeframe } = args
+
+        if (assetIds.length === 0) return { data: null }
+
+        const responseData: { assetId: AssetId; historyData: HistoryData[] }[] = await Promise.all(
```

Comment:
this is going to have an unintended consequence of beating up coingecko - if we request market data for `[a, b, c]` then request market data for `[a, b, c, d]` we're going to refetch `[a, b, c]` unnecessarily

this fn should

a) take in the array of assets
b) call another queryFn for each asset, which makes the network request and returns the data, but doesn't upsert it to the store, which will cache correctly on a per asset basis
c) aggregate all the market data and upsert in one hit

Reply:
excellent point, as discussed i've added axios caching to avoid nested rtk endpoints
3437e68d18e0fe7189b9cb7fe1b1f629397a3a62